### ISSUE TITLE
[BUG] cant run element.requestFullscreen() without user interaction

### ISSUE THREAD

**Issue by saty9** [obsproject/obs-browser#224]
## Platform

Operating system and version:
Ubuntu 18.04.4 LTS
(also reproduced on windows 10)
OBS Studio version:
installed using ubuntu store snap version 25.0.7 rev 1063

## Expected Behavior

A page that runs videoelement.requestFullscreen() should have that element go fullscreen

## Current Behavior

The action is blocked with the message:

``'requestFullscreen' on 'Element': API can only be initiated by a user gesture.``

## Steps to Reproduce

https://jsfiddle.net/saty9/xpntf1qc/3/
 Should have enough to reproduce a page with the same issue. I believe it cant be hosted from a local file for testing as that may bypass the restriction (I haven't checked this)

## Additional information

The video is displaying data from a webrtc stream in my usecase

**Comment by Lange** [comment]
Can you elaborate on what the use case for the fullscreen API is in a browser source? I admit I don't even know what invoking fullscreen would do on a webpage loaded in this way. What does it let you achieve that other APIs don't in this context?

**Comment by dodgepong** [comment]
I suspect they are trying to fullscreen a video player, perhaps.

**Comment by Lange** [comment]
I agree that's a likely scenario. It should be noted that, in the context of a browser source, a visually similar outcome can be reached via something like:

```css
video {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%
}
```

That said, I can't immediately think of any security reasons for not allowing the fullscreen API to be freely invoked in a browser source, and `obs-browser` already makes a similar exception for the general media playback permissions: https://github.com/obsproject/obs-browser/issues/105

**Comment by WizardCM** [comment]
After testing this myself, the behaviour is correct in terms of behaving like a web browser - that is, the fullscreen API fails if you have not yet interacted with the site, but succeeds if you have. I've confirmed Microsoft Edge 91 does the same thing as obs-browser.

I can however see a use case for potentially sending a dummy interaction on load to make the page think a user has interacted with it, to allow for APIs such as this to succeed without actual user interaction (via the Interact dialog). The downside of doing this is there are certain APIs that *shouldn't* be run without actual user interaction.
